File: C++ solutions/128. Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(!nums.size()) return 0;
        
        map<int,int>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }
        vector<int>v;
        for(auto k:mp) v.push_back(k.first);

        int cnt=1,ans=0;
        for(int i=1;i<v.size();i++){
            if(v[i]-v[i-1]==1) cnt++;
            else{
                ans=max(ans,cnt);
                cnt=1;
            }
        }
        ans=max(ans,cnt);
        return ans;
    }
};

File: JavaScript solutions/128. Longest Consecutive Sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if (nums.length === 0) return 0;

    let mp = new Map();
    for (let i = 0; i < nums.length; i++) {
        mp.set(nums[i], (mp.get(nums[i]) || 0) + 1);
    }
    
    let v = Array.from(mp.keys()).sort((a, b) => a - b);

    let cnt = 1;
    let ans = 0;
    for (let i = 1; i < v.length; i++) {
        if (v[i] - v[i - 1] === 1) {
            cnt++;
        } else {
            ans = Math.max(ans, cnt);
            cnt = 1;
        }
    }
    ans = Math.max(ans, cnt);
    return ans;
};